### PR TITLE
Add support multiple attribute in `HelpOptions` select type

### LINKED ISSUE #29903: `HelperOptions` select multiple not work
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

`HelperOptions` select multiple not work

image

### Expected behavior

Able to add multiple attribute for select type

image

### Steps to reproduce

Edit `AdminGroupsControllerCore::__construct`

Add multiple attribute : 

```php
'PS_UNIDENTIFIED_GROUP' => [
         'multiple' => true,
         'class' => 'chosen' ,
         'title' => $this->trans('Visitors group', [], 'Admin.Shopparameters.Feature'),
         'desc' => $this->trans('The group defined for your un-identified visitors.', [], 'Admin.Shopparameters.Help'),
         'cast' => 'intval',
         'type' => 'select',
          'list' => $groups,
         'identifier' => 'id_group',
 ],
```

See the error : 

image

### PrestaShop version(s) where the bug happened

1.7.8.7, develop

### PHP version(s) where the bug happened

N/A

### If your bug is related to a module, specify its name and its version

N/A

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add support multiple attribute in `HelpOptions` select type.
| Type?             | bug fix
| Category?         | CO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/29903
| Related PRs       | N/A
| How to test?      | N/A
| Possible impacts? | N/A

### CODE REVIEW

File: admin-dev/themes/default/template/helpers/options/options.tpl [hunk 1]
```diff
@@ -131,7 +131,7 @@
 									{if $field['type'] == 'select'}
 										<div class="col-lg-9">
```
Comment (lartist) [inline admin-dev/themes/default/template/helpers/options/options.tpl:132]: Hi @PululuK. 
`isset($field['multiple']) && $field['multiple']` can be replaced by `!empty($field['multiple'])`.
According to php.net: "A variable is considered empty if it does not exist or if its value equals false"

**Review by FabienPapet — DISMISSED** [review]

File: admin-dev/themes/default/template/helpers/options/options.tpl [hunk 2]
```diff
@@ -131,7 +131,7 @@
...
 											{if $field['list']}
-												<select class="form-control fixed-width-xxl {if isset($field['class'])}{$field['class']}{/if}" name="{$key}"{if isset($field['js'])} onchange="{$field['js']}"{/if} id="{$key}" {if isset($field['size'])} size="{$field['size']}"{/if}>
+												<select class="form-control fixed-width-xxl {if isset($field['class'])}{$field['class']}{/if}" name="{$key}"{if isset($field['js'])} onchange="{$field['js']}"{/if} id="{$key}" {if isset($field['size'])} size="{$field['size']}"{/if} {if isset($field['multiple']) && $field['multiple']} multiple="multiple"{/if}>
```
Comment (kpodemski) [inline admin-dev/themes/default/template/helpers/options/options.tpl]:
```suggestion
												<select class="form-control fixed-width-xxl {if isset($field['class'])}{$field['class']}{/if}" name="{$key}"{if isset($field['js'])} onchange="{$field['js']}"{/if} id="{$key}" {if isset($field['size'])} size="{$field['size']}"{/if} {if !empty($field['multiple'])} multiple="multiple"{/if}>
```

**Review by yanmakouf — DISMISSED** [review]

File: admin-dev/themes/default/template/helpers/options/options.tpl (hunk 1 above)
Comment (PululuK) [reply admin-dev/themes/default/template/helpers/options/options.tpl:132]: Thanks @lartist

**Review by kpodemski — APPROVED** [review]

File: admin-dev/themes/default/template/helpers/options/options.tpl (hunk 2 above)
Comment (PululuK) [reply admin-dev/themes/default/template/helpers/options/options.tpl]: Thanks @kpodemski

**Review by marsaldev — APPROVED** [review]

**Review by MhiriFaten — COMMENTED** [review]
Hello @PululuK ,

I checked this PR. As you can see in the screen bellow, it seems that the issue is not fixed.
The behavior with PR is different compared to the expected behavior described in the issue.

image


What do you think @PrestaShop/prestashop-core-developers ! ?

Thank you

**Comment by PululuK** [comment]
Hi @MhiriFaten 
It is work fine ... This is the expected behavior :) 
You can see now can select multiple options, if you want to have the same result like my screen, you need to add this attribute `'class' => 'chosen' ` (I just update the issue).

Thanks

**Review by MhiriFaten — COMMENTED** [review]
Hi @PululuK 

I rechecked the PR and as you can see in the screen record :

[Untitled_ Nov 2, 2022 8_01 AM.webm](https://user-images.githubusercontent.com/111756615/199436958-189b4ec4-5be5-415c-bbe1-c372b7e32ebe.webm)

The select multiple still not work even when I select multiple groups, the display after the save shows only one choice.

**Comment by kpodemski** [comment]
@MhiriFaten this change works as expected, it's about rendering the field, not saving the option

@PululuK change is not about saving the data

**Review by nicosomb — APPROVED** [review]

**Comment by MhiriFaten** [comment]
Thanks @kpodemski for your explanation.

Ping @PrestaShop/product-team what do you think about this behavior?

**Comment by kpodemski** [comment]
@MhiriFaten 

again, it is not about behavior change :) you can **render select field that supports multi-selection**, and that's it

That's just the first step for developers, then they need to make it work with data persistence 👍

**Comment by kpodemski** [comment]
@PrestaShop/qa-team 

this is a developer feature, can we agree on QA by a dev?

**Comment by hibatallahAouadni** [comment]
@kpodemski Yes, if one of the developers reproduce and verified the related issue! Yessss!
So, please update the labels of the issue and this PR :pray: 

Thanks in advance!

**Comment by kpodemski** [comment]
Thanks @PululuK :)

**Comment by PululuK** [comment]
Thanks @kpodemski

**Pull request merged**